### PR TITLE
Acts as slugged

### PR DESCRIPTION
**Pull request by alanvardy**
Adds slugs to effective posts using effective resources as a dependency

### CODE REVIEW

File: app/models/effective/post.rb [hunk 1]
```diff
@@ -67,6 +70,10 @@ class Post < ActiveRecord::Base
       scope
     }
 
+    def to_param
+      slug || "#{id}-#{title.parameterize}"
+    end
+
```
Comment (alanvardy) [inline app/models/effective/post.rb:76]: I don't know if having this fallback is a good idea, but it sure helps with the old data.

File: effective_posts.gemspec [hunk 2]
```diff
@@ -23,4 +23,5 @@ Gem::Specification.new do |s|
   s.add_dependency 'effective_ckeditor'
   s.add_dependency 'effective_datatables', '>= 4.0.0'
   s.add_dependency 'effective_regions'
+  s.add_dependency 'effective_resources'
```
Comment (alanvardy) [inline effective_posts.gemspec:26]: We need to include this dependency right?

File: app/views/admin/posts/_form.html.haml [hunk 3]
```diff
@@ -1,5 +1,8 @@
 = effective_form_with(model: post, url: post.persisted? ? effective_posts.admin_post_path(post.id) : effective_posts.admin_posts_path) do |f|
   = f.text_field :title, hint: 'The title of your post.'
+  - if f.object.persisted? || f.object.errors.include?(:slug)
+    - current_url = (effective_pages.post_url(f.object) rescue nil)
```
Comment (matt-riemer) [inline app/views/admin/posts/_form.html.haml:4]: effective_posts.post_url(f.object)

File: app/views/admin/posts/_form.html.haml [hunk 4]
```diff
@@ -1,5 +1,8 @@
...
+    = f.text_field :slug, hint: "The slug controls this page's internet address. Be careful, changing the slug will break links that other websites may have to the old address.<br>#{('This page is currently reachable via ' + link_to(current_url.gsub(f.object.slug, '<strong>' + f.object.slug + '</strong>').html_safe, current_url)) if current_url }".html_safe
```
Comment (matt-riemer) [inline app/views/admin/posts/_form.html.haml]: could you just `gsub("page", "post")` the text here :)

Comment (matt-riemer) [reply app/views/admin/posts/_form.html.haml]: "The slug controls this post's internet address..."

File: effective_posts.gemspec (hunk 2 above)
Comment (matt-riemer) [reply effective_posts.gemspec:26]: good catch, yes thanks!

File: app/views/admin/posts/_form.html.haml [hunk 5]
```diff
@@ -1,5 +1,8 @@
 = effective_form_with(model: post, url: post.persisted? ? effective_posts.admin_post_path(post.id) : effective_posts.admin_posts_path) do |f|
   = f.text_field :title, hint: 'The title of your post.'
```
Comment (matt-riemer) [inline app/views/admin/posts/_form.html.haml:2]: Could you also add this same stuff to the form at 'app/views/effective/posts/_form`

We have an admin and a front facing form. Both can use the slugs I guess :)

**Comment by matt-riemer** [comment]
Hey not included in this is:

Please add "t.string :slug" to the db/migrate/01_create_effective_posts.rb.erb file

You rock thanks :D

File: app/views/admin/posts/_form.html.haml (hunk 4 above)
Comment (alanvardy) [reply app/views/admin/posts/_form.html.haml]: Done!

**Comment by alanvardy** [comment]
It is included! Are you not seeing it on your end?

Screenshot from 2019-06-20 08-53-28

**Comment by matt-riemer** [comment]
Thanks Alan!

File: app/views/admin/posts/_form.html.haml (hunk 5 above)
Comment (alanvardy) [reply app/views/admin/posts/_form.html.haml:2]: Oh cool! Added.

**Pull request merged**